src/app/metadata/features/dataset-details/dataset-details.component.html

```diff
@@ -128,13 +128,20 @@ <h4 class="inline-flex items-center">
                   <span
                     class="bg-primary/15 text-primary inline-flex items-center rounded p-1 text-4xl"
                     ><mat-icon inline>article</mat-icon></span
-                  >&nbsp;{{ pub.title }}
+                  >&nbsp;{{ pub.title ?? 'Unknown publication title' }}
                 </h4>
-                <p class="text-base"><strong>Author(s):</strong> {{ pub.author }}</p>
-                <p class="text-base"><strong>Journal:</strong> {{ pub.journal }}</p>
-                <p class="text-base"><strong>Year:</strong> {{ pub.year }}</p>
+                <p class="text-base">
+                  <strong>Author(s):</strong> {{ pub.author ?? 'Unknown authors' }}
+                </p>
+                @if (pub.journal) {
+                  <p class="text-base"><strong>Journal:</strong> {{ pub.journal }}</p>
+                }
+                @if (pub.year) {
+                  <p class="text-base"><strong>Year:</strong> {{ pub.year }}</p>
+                }
+                <p class="text-base"><strong>DOI:</strong> {{ pub.doi }}</p>
                 @for (
-                  par of pub.abstract.split('\\n');
+                  par of pub.abstract?.split('\\n');
```

Comment:
I would wrap this for loop into an `if (pub.abstract)` because it's an implementation detail how the Angular compiler deals with an undefined value in this place, I would not want to rely on that (in normal JavaScript, you cannot iterate over undefined).

Ideally, paragraphs should be also trimmed and empty paragraphs be filtered out, but not in this hotfix because this probably should be done in a pipe.

Reply:
Now that I think about it, would this not be one of these 'no function calls in template'? @SilverLinings89

Reply:
Yes, I think we should either precompute the array and store it in a property or use a pipe. Doing so, we could then also do the trimming and filtering.

Reply:
We should do this in a different PR.